src/frequenz/sdk/actor/power_distributing/_component_status.py

```diff
@@ -0,0 +1,109 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Classes to track the status of components in the microgrid."""
+
+
+import enum
+import typing
+from abc import ABC, abstractmethod
+from collections import abc
+from dataclasses import dataclass
+
+from frequenz.channels import Receiver, Sender
+
+
+@dataclass
```

Comment:
can this dataclass be `frozen=True, kw_only=True` or there is a reason to not do it?

Reply:
probably, but out of scope for this PR.